src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FootnoteService.cs

```diff
@@ -130,7 +130,24 @@ public class FootnoteService : IFootnoteService
                 .OnSuccess(_ => _statisticsPersistenceHelper.CheckEntityExists<Footnote>(footnoteId)
                 .OnSuccessVoid(async footnote =>
                 {
-                    await _footnoteRepository.DeleteFootnote(releaseId, footnote.Id);
+                    if (await _footnoteRepository.IsFootnoteExclusiveToRelease(releaseId: releaseId,
+                            footnoteId: footnoteId))
+                    {
+                        await _footnoteRepository.DeleteFootnote(releaseId: releaseId,
+                            footnoteId: footnote.Id);
+                    }
+                    else
+                    {
+                        // TODO EES-2979 Remove this once a footnote can only belong to one release
+                        // It's only be possible to reach this code if editing a very old draft release amendment,
+                        // created circa September-2020 when we began copying footnotes when creating new versions.
+                        await _footnoteRepository.DeleteReleaseFootnoteLink(releaseId: releaseId,
+                            footnoteId: footnoteId);
+
+                        // TODO Because the footnote isn't deleted this could be leaving footnote links to subjects or
+                        // filters/filter groups/filter items of subjects that only belong to this release.
+                        // It seems unlikely we will need to fix this, but prioritise EES-2979 instead. 
+                    }
```

Comment:
Couldn't all this logic be handled internally within `FootnoteRepository.DeleteFootnote` (similar to `DeleteFootnotesBySubject`)?

If we did that, we could just keep all the logic in there and just call `DeleteFootnote`.

Reply:
Thanks, I actually moved this kind of logic out of `FootnoteRepository.DeleteFootnote` to make sure the only thing that was doing was actually deleting a foonote, but I agree it's a bit cleaner this way so I've moved it back.

It will be a lot cleaner all over the place once we've done EES-2979.